app/helpers/error_messages_helper.rb

```diff
@@ -0,0 +1,42 @@
+# encoding: utf-8
+# Nifty Generators - https://github.com/ryanb/nifty-generators
+#
+# copy from Nifty Generators,
+# markup optimized for bootstrap
+#
+# Usage:
+#    <%= f.error_messages %>
+#    <%= error_messages_for(@object) %>
+#
+# Markup aliged with bootstrap v4
+module ErrorMessagesHelper
```

Comment:
Do we need this? This was something that Rails had support for it and we removed because it was too magical, I really don't want to add it back in an example application.

Reply:
It's currently unused in the example application. I thought it would be a good idea to add it as a reference. You never know, if some people find this useful. It's a working example of a side project of mine. That's why I add comments and kudos to [ryanb/nifty-generators](https://github.com/ryanb/nifty-generators). Of course will remove it, if it bothers you.

Imho, Simple Forms build in `f.error_notification` has a very minimalistic and simple output. The `ErrorMessagesHelper` / `f.error_messages`-helper with Bootstrap v4 markup would just provide a reference for some more extended error messages. Ready to use. Example attached.

![screen_shot_2018-06-06_at_11_00_49](https://user-images.githubusercontent.com/26371/41028268-53ed1b2c-6979-11e8-9fc4-f23c24053de2.png)

Reply:
Fwiw, I think supporting both options is most sensible for this library, since not everyone using it must necessarily keep using inline errors and it also helps when migrating custom non-simple_form elements' styling.